src/Core/UserManager.cs

```diff
@@ -527,15 +541,36 @@ public virtual Task<TUser> FindByIdAsync(string userId)
         /// <returns>
         /// The <see cref="Task"/> that represents the asynchronous operation, containing the user matching the specified <paramref name="userName"/> if it exists.
         /// </returns>
-        public virtual Task<TUser> FindByNameAsync(string userName)
+        public virtual async Task<TUser> FindByNameAsync(string userName)
         {
             ThrowIfDisposed();
             if (userName == null)
             {
                 throw new ArgumentNullException(nameof(userName));
             }
             userName = NormalizeKey(userName);
-            return Store.FindByNameAsync(userName, CancellationToken);
+
+            var user = await Store.FindByNameAsync(userName, CancellationToken);
+
+            // Need to potentially check all keys
+            if (user == null && Options.Stores.ProtectPersonalData)
+            {
+                var keyRing = _services.GetService<ILookupProtectorKeyRing>();
+                var encryptor = _services.GetService<ILookupProtector>();
```

Comment:
replace encrypt => protector, also in IEncryptedUserStore => IProtectedUserStore